std/math/exponential.d

```diff
@@ -826,37 +826,211 @@ if (isUnsigned!F && isUnsigned!G && isUnsigned!H)
         alias DoubleT = AliasSeq!(void, ushort, uint, void, ulong)[T.sizeof];
     }
 
-    static T mulmod(T a, T b, T c)
+    static T mulmod(T)(T a,T b,T c)
     {
         static if (T.sizeof == 8)
         {
-            static T addmod(T a, T b, T c)
+            if (c <= 0x100000000)
             {
-                b = c - b;
-                if (a >= b)
-                    return a - b;
-                else
-                    return c - b + a;
+                T result=a*b;
+                return result%c;
+            }
+            version (D_InlineAsm_X86_64)
+            {
+                ulong low=void;
+                ulong high=void;
+
+                asm pure @trusted nothrow @nogc
+                {
+                    mov RAX,a;
+                    mul b;
+                    mov low,RAX;
+                    mov high,RDX;
+                }
+
+                if (high >= c)
+                {
+                    high%=c;
+                }
+
+                if (high == 0)
+                {
+                    return low%c;
+                }
+
+                asm pure @trusted nothrow @nogc
+                {
+                    mov RAX,high;
+                    xor RDX,RDX;
+                    div c;
+                    mov high,RDX;
+
+                    mov RAX,low;
+                    mov RDX,high;
+                    div c;
+                    mov low,RDX;
+                }
+
+                return low;
+            }
+            else version (LDC)
+            {
+                uint low,high;
+
+                asm pure @trusted nothrow @nogc
+                {
+                    "mull %2"
+                    :"=a"(low),"=d"(high)
+                    :"r"(b),"0"(a)
+                    :"cc";
+                }
+
+                if (high >= c)
+                {
+                    high%=c;
+                }
+
+                if (high == 0)
+                {
+                    return low%c;
+                }
+
+                asm pure @trusted nothrow @nogc
+                {
+                    "divl %2"
+                    :"=a"(low),"=d"(high)
+                    :"r"(c),"0"(low),"1"(high)
```

Comment:
This can be simplified with `+` constraint as this is reading _and_ writing to `low` and `high`.

```suggestion
                    :"+a"(low),"+d"(high)
                    :"r"(c)
```